packages/design-system/src/components/Tabs/TabPanel.tsx

```diff
@@ -1,67 +1,64 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import classnames from 'classnames';
 
-export function TabPanel(props) {
-  const classes = classnames('ds-c-tabs__panel', props.className);
-
-  return (
-    <div
-      aria-labelledby={props.tabId}
-      aria-hidden={String(!props.selected)}
-      aria-disabled={String(!!props.disabled)}
-      className={classes}
-      id={props.id}
-      role="tabpanel"
-    >
-      {props.children}
-    </div>
-  );
-}
-
-// Set component name to make child.type.displayName available to other components (eg. Tab)
-TabPanel.displayName = 'TabPanel';
-
-TabPanel.defaultProps = {
-  selected: false,
-};
-
-TabPanel.propTypes = {
-  children: PropTypes.node.isRequired,
+export interface TabPanelProps {
+  children: React.ReactNode;
   /**
    * Additional classes to be added to the root element.
    */
-  className: PropTypes.string,
+  className?: string;
   /**
    * A unique `id`, to be used on the rendered panel element.
    */
-  id: PropTypes.string.isRequired,
-  selected: PropTypes.bool,
-  disabled: PropTypes.bool,
-  /* eslint-disable react/no-unused-prop-types */
+  id: string;
+  selected?: boolean;
+  disabled?: boolean;
   /**
    * The associated tab's label. Only applicable when the panel is a
    * child of `Tabs`.
    */
-  tab: PropTypes.string,
+  tab?: string;
   /**
    * Additional classes for the associated tab. Only applicable when the panel
    * is a child of `Tabs`.
    */
-  tabClassName: PropTypes.string,
+  tabClassName?: string;
   /**
    * The associated tab's `href`. Only applicable when the panel is a
    * child of `Tabs`.
    */
-  tabHref: PropTypes.string,
-  /* eslint-enable react/no-unused-prop-types */
+  tabHref?: string;
   // tabId is actually required, but it's not marked here since we generate
   // this id within the Tabs component. Otherwise React will yell at you even
   // though it's ultimately being passed in.
   /**
    * The `id` of the associated `Tab`. Used for the `aria-labelledby` attribute.
    */
-  tabId: PropTypes.string,
+  tabId?: string;
+}
+
+export function TabPanel(props: TabPanelProps): JSX.Element {
```

Comment:
Wondering, for consistency could we write TS functions like:

`export const TabPanel = (props: TabPanelProps) => {`

And perhaps leaving off the return type (unless it's more specific like a string/bool/int) and allowing TS to infer it in these cases?  It seems like strongly typing some of these might cause issues down the line, especially with JSX.Element as it's not as permissive as React.ReactNode.

I am mainly suggesting this because I prefer this syntax, feel free to ignore 🖖

Reply:
@zarahzachz @pwolfert as I commented here and below, maybe we can agree on a function syntax and when to use return values as a way of structuring a new team norm? Thoughts?

Reply:
I'm happy to convert to the `export cont TabPanel = () => {}` syntax for all functions

With regard to the return type, I kind of agree that JSX.Element is a strict typing and could potentially cause problem eslint throws a warning if a function doesn't have a return type. 😞

Reply:
Right on! I think that's one of the eslint warnings we should turn off in the future

Reply:
Agreed